kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+    public static final String RECONCILER_GROUPS_TO_REROUTE_PARM_NM = "RECONCILER_GROUPS_TO_REROUTE";
+    
+	private boolean hasReconciler(ProcurementCardDefault procurementCardDefault) {
+        boolean retCode = true;
+        if (ObjectUtils.isNull(procurementCardDefault) ||
+            ObjectUtils.isNull(procurementCardDefault.getReconcilerGroupId()) ||
+            ObjectUtils.isNull(procurementCardDefault.getCardHolderSystemId())) {
+            retCode = false;
+        }
+        else {
+            List<String> groupMembers = new ArrayList<String>();
+            String reconcilerGroupId = procurementCardDefault.getReconcilerGroupId();
+            groupMembers = SpringContext.getBean(GroupService.class).getMemberPrincipalIds(reconcilerGroupId);
+            if (groupMembers.isEmpty() ||
+                (groupMembers.size() == 1 &&
+                 groupMembers.get(0).equals(procurementCardDefault.getCardHolderSystemId()))) {
+                retCode = false;
+            }
+        }
+        return retCode;
+    }
+	
+    /**
+     * This method looks for ProcurementCardDocuments that are in route status at the "AccountFullEdit" route node and routed to the error account FO. 
+     * Then checks for a valid reconciler and if found, reroutes the document back to the Reconciler.
+     * 
+     * 
+     * @return True if the routing was performed successfully.  A runtime exception will be thrown if any errors occur while routing.
+     * 
+     * @see org.kuali.kfs.fp.batch.service.ProcurementCardCreateDocumentService#rerouteProcurementCardDocuments(java.util.List)
+     */
+	@Override
+    public boolean rerouteProcurementCardDocuments() {
+        
+        // check for reconciler group rerouting
+        boolean groupRerouting = parameterService.getParameterValueAsBoolean(ProcurementCardRerouteDocumentsStep.class, REROUTE_PCDO_DOCUMENTS_IND_PARM_NM);
+        
+        Collection<ProcurementCardDocument> documents = null;
+        documents = retrieveUAProcurementCardDocumentsToRoute(KewApiConstants.ROUTE_HEADER_ENROUTE_CD);
+        
+        if ( LOG.isInfoEnabled() ) {
+            LOG.info("PCards to Reroute: " + new Integer(documents.size()).toString());
+        }
+        for (ProcurementCardDocument pcardDocument : documents) {
+        
+        	String pcardDocumentId = pcardDocument.getObjectId();
+            //get document route node
+            List<RouteNodeInstance> routeNodeInstances = pcardDocument.getDocumentHeader().getWorkflowDocument().getCurrentRouteNodeInstances();
+            String node = routeNodeInstances.get(0).getName();
+            //get Procurement Card Default information
+            ProcurementCardDefault procurementCardHolderDetail = getProcurementCardDefault(pcardDocument.getProcurementCardHolder().getTransactionCreditCardNumber());
+            boolean hasReconciler = hasReconciler(procurementCardHolderDetail);
```

Comment:
While stepping through with the debugger, it is (was) helpful to have a 'hasReconciler' boolean field to inspect prior to the evaluation of the 'if' statement.